types/chainLock.go

```diff
@@ -0,0 +1,123 @@
+package types
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/tendermint/tendermint/crypto"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+type ChainLock struct {
+	CoreBlockHeight    uint32          `json:"core_block_height,string,omitempty"`   // height of Chain Lock.
```

Comment:
`BlockHeigth`?

Reply:
There are 2 block heights, one core and one platform, expressing which one I think makes this more easy to understand.

Reply:
But it's a field in ChainLock structure, so it already namespaced by Core, right? It's like when you have an object `core` you don't name properties like `core.coreBlockHeight`, `core.coreBestChainLock` or something like this.